src/App.tsx

```diff
@@ -1,18 +1,30 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
+const items: string[] = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [itemsPerPage, setItemsPerPage] = useState(3);
+
+  const totalPages = items.length;
+  const firstVisibleItem = (currentPage * itemsPerPage) - itemsPerPage;
+  const lastVisibleItem = (firstVisibleItem + itemsPerPage - 1) <= totalPages
+    ? firstVisibleItem + itemsPerPage
+    : items.length;
+
+  const visibleArray = [...items].slice(firstVisibleItem, lastVisibleItem);
```

Comment:
```suggestion
  const visibleArray = items.slice(firstVisibleItem, lastVisibleItem);
```
looks redundant, the slice doesn't mutate the original array

Reply:
You're rigth! Done/